Tools/cases_generator/parsing.py

```diff
@@ -71,6 +71,9 @@ class Block(Node):
 class StackEffect(Node):
     name: str = field(compare=False)  # __eq__ only uses type, cond, size
     type: str = ""  # Optional `:type`
+    # Optional `(type, aux)`
+    type_prop: None | tuple[str, None | str] = \
+        field(default_factory=lambda: None, init=True, compare=False, hash=False)
```

Comment:
I find it slightly jarring that this is a positional parameter inserted in the middle, whereas earlier, in StackItem, you add it at the end as a keyword parameter. Maybe use the same style in both places?